controllers/ingressnodefirewall_controller_rules_test.go

```diff
@@ -0,0 +1,753 @@
+								{
+									Order: 10,
+									ProtocolRule: &infv1alpha1.IngressNodeFirewallProtoRule{
+										Ports: "80",
+									},
+									Protocol: "tcp",
+									Action:   "allow",
+								},
+							},
+						},
+						{
+							SourceCIDRs: []string{"10.0.1.0"},
+							FirewallProtocolRules: []infv1alpha1.IngressNodeFirewallProtocolRule{
+								{
+									Order: 10,
+									ProtocolRule: &infv1alpha1.IngressNodeFirewallProtoRule{
+										Ports: "81",
+									},
+									Protocol: "tcp",
+									Action:   "deny",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		// TC7 - merging rules with multiple CIDRs.
+		{
+			inSpecs: []infv1alpha1.IngressNodeFirewallSpec{
+				{
+					Ingress: []infv1alpha1.IngressNodeFirewallRules{
+						{
+							SourceCIDRs: []string{"10.0.0.0", "10.0.1.0"},
+							FirewallProtocolRules: []infv1alpha1.IngressNodeFirewallProtocolRule{
+								{
+									Order: 10,
+									ProtocolRule: &infv1alpha1.IngressNodeFirewallProtoRule{
+										Ports: "80",
+									},
+									Protocol: "tcp",
+									Action:   "allow",
+								},
+							},
+						},
+					},
+					Interfaces: &[]string{"eth0"},
+				},
+				{
+					Ingress: []infv1alpha1.IngressNodeFirewallRules{
+						{
+							SourceCIDRs: []string{"10.0.0.0", "10.0.1.0"},
+							FirewallProtocolRules: []infv1alpha1.IngressNodeFirewallProtocolRule{
+								{
+									Order: 20,
+									ProtocolRule: &infv1alpha1.IngressNodeFirewallProtoRule{
+										Ports: "81",
+									},
+									Protocol: "tcp",
+									Action:   "deny",
+								},
+							},
+						},
+					},
+					Interfaces: &[]string{"eth0"},
+				},
+			},
+			outSpec: infv1alpha1.IngressNodeFirewallNodeStateSpec{
+				InterfaceIngressRules: map[string][]infv1alpha1.IngressNodeFirewallRules{
+					"eth0": {
+						{
+							SourceCIDRs: []string{"10.0.0.0"},
+							FirewallProtocolRules: []infv1alpha1.IngressNodeFirewallProtocolRule{
+								{
+									Order: 10,
+									ProtocolRule: &infv1alpha1.IngressNodeFirewallProtoRule{
+										Ports: "80",
+									},
+									Protocol: "tcp",
+									Action:   "allow",
+								},
+								{
+									Order: 20,
+									ProtocolRule: &infv1alpha1.IngressNodeFirewallProtoRule{
+										Ports: "81",
+									},
+									Protocol: "tcp",
+									Action:   "deny",
+								},
+							},
+						},
+						{
+							SourceCIDRs: []string{"10.0.1.0"},
+							FirewallProtocolRules: []infv1alpha1.IngressNodeFirewallProtocolRule{
+								{
+									Order: 10,
+									ProtocolRule: &infv1alpha1.IngressNodeFirewallProtoRule{
+										Ports: "80",
+									},
+									Protocol: "tcp",
+									Action:   "allow",
+								},
+								{
+									Order: 20,
+									ProtocolRule: &infv1alpha1.IngressNodeFirewallProtoRule{
+										Ports: "81",
+									},
+									Protocol: "tcp",
+									Action:   "deny",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
```

Comment:
maybe a test case for same CIDR, order different protocol ?

Reply:
done